backend/src/modules/transfer/utils/validateDigitalFileList.ts

```diff
@@ -0,0 +1,94 @@
+import type { Buffer } from "node:buffer";
+import { HttpError, HTTP_STATUS_CODES } from "@bcgov/citz-imb-express-utilities";
+import xlsx from "xlsx";
+
+type Data = {
+	buffer: Buffer;
+	accession: string;
+	application: string;
+};
+
+export const validateDigitalFileList = async ({
+	buffer,
+	accession,
+	application,
+}: Data): Promise<void> => {
+	// Check file type
+	const bufferType = buffer.toString("hex", 0, 4);
+	const isXlsx = bufferType === "504b0304"; // Zip header for .xlsx files
+
+	let isJson = false;
+
+	try {
+		// Attempt to parse buffer as JSON to confirm it's a valid JSON file
+		JSON.parse(buffer.toString("utf-8"));
+		isJson = true;
+	} catch {
+		isJson = false;
+	}
+
+	if (!isXlsx && !isJson) {
+		throw new HttpError(
+			HTTP_STATUS_CODES.BAD_REQUEST,
+			"The Digital File List must be a .xlsx or .json file.",
+		);
+	}
+
+	if (isXlsx) {
+		// Handle XLSX validation
+		const workbook = xlsx.read(buffer, { type: "buffer" });
+		const coverPage = workbook.Sheets["COVER PAGE"];
+
+		if (!coverPage) {
+			throw new HttpError(
+				HTTP_STATUS_CODES.BAD_REQUEST,
+				'The Digital File List must contain a "COVER PAGE" sheet.',
+			);
+		}
+
+		const cellB2 = coverPage.B2?.v;
+		const cellB3 = coverPage.B3?.v;
+		const cellB4 = coverPage.B4?.v;
+
+		if (cellB2 === "") {
+			throw new HttpError(
+				HTTP_STATUS_CODES.BAD_REQUEST,
+				"The Digital File List is missing the Last Revised Date.",
+			);
+		}
+
+		if (cellB3 !== accession || cellB4 !== application) {
+			throw new HttpError(
+				HTTP_STATUS_CODES.BAD_REQUEST,
+				"The Digital File List is missing or has incorrect accession and application numbers.",
```

Comment:
Noting that this fails if the excel file has these fields as numbers instead of strings.
Log from my testing:
```
cellB3:  1234567 . Typeof:  number
accession:  1234567 . Typeof:  string
cell B4:  7654321 . Typeof:  number
application:  7654321 . Typeof:  string
The Digital File List is missing or has incorrect accession and application numbers.
```

Reply:
I dont think there will ever be a case where they are number types. Should always be strings

Reply:
are we getting the excel file from us or would a use be able to edit it at any point?
- If a user can touch it we cant guarantee that they wont change that. 
- But if it is just from us and never in the users hand I am happy to ignore this.

Reply:
But it isnt always read as a string. With my test I went into the file and changed the cell type to a number and this is the error I got.

Reply:
I see. I'll cast it to a string